src/tools/rustdoc-js-not-std/tester.js

```diff
@@ -0,0 +1,365 @@
+            return;
+        }
+
+        var out_folder = "build/" + toolchain + "/stage" + stage + "/tests/rustdoc-js-not-std/" +
+                         test_name;
+
+        var ret = build_docs(out_folder, rustdoc_path, rust_file);
+        if (ret.length > 0) {
+            console.error("FAILED");
+            console.error(ret);
+            errors += 1;
+            return;
+        }
+
+        var [loaded, index] = load_files(out_folder, test_name);
+        var loadedFile = loadContent(readFile(TEST_FOLDER + file) +
+                               'exports.QUERY = QUERY;exports.EXPECTED = EXPECTED;');
+        const expected = loadedFile.EXPECTED;
+        const query = loadedFile.QUERY;
+        const filter_crate = loadedFile.FILTER_CRATE;
+        const ignore_order = loadedFile.ignore_order;
+        const exact_check = loadedFile.exact_check;
+        const should_fail = loadedFile.should_fail;
+        var results = loaded.execSearch(loaded.getQuery(query), index);
+        var error_text = [];
+        for (var key in expected) {
+            if (!expected.hasOwnProperty(key)) {
+                continue;
+            }
+            if (!results.hasOwnProperty(key)) {
+                error_text.push('==> Unknown key "' + key + '"');
+                break;
+            }
+            var entry = expected[key];
+            var prev_pos = -1;
+            for (var i = 0; i < entry.length; ++i) {
+                var entry_pos = lookForEntry(entry[i], results[key]);
+                if (entry_pos === null) {
+                    error_text.push("==> Result not found in '" + key + "': '" +
+                                    JSON.stringify(entry[i]) + "'");
+                } else if (exact_check === true && prev_pos + 1 !== entry_pos) {
+                    error_text.push("==> Exact check failed at position " + (prev_pos + 1) + ": " +
+                                    "expected '" + JSON.stringify(entry[i]) + "' but found '" +
+                                    JSON.stringify(results[key][i]) + "'");
+                } else if (ignore_order === false && entry_pos < prev_pos) {
+                    error_text.push("==> '" + JSON.stringify(entry[i]) + "' was supposed to be " +
+                                    " before '" + JSON.stringify(results[key][entry_pos]) + "'");
+                } else {
+                    prev_pos = entry_pos;
+                }
+            }
+        }
+        if (error_text.length === 0 && should_fail === true) {
+            errors += 1;
+            console.error("FAILED");
+            console.error("==> Test was supposed to fail but all items were found...");
+        } else if (error_text.length !== 0 && should_fail === false) {
+            errors += 1;
+            console.error("FAILED");
+            console.error(error_text.join("\n"));
+        } else {
+            // In this case, we remove the docs, no need to keep them around.
+            remove_docs(out_folder);
```

Comment:
Removing the built docs every time means we have to rebuild them every time, even if the test crate never changed.

This is one of the reasons i wanted to move the logic into `compiletest`, because it can do that timestamp checking to build the docs only when the test (or rustdoc) has changed.